io/src/lib.rs

```diff
@@ -373,4 +424,53 @@ mod tests {
         assert_eq!(read, 2);
         assert_eq!(&buf, "16".as_bytes())
     }
+
+    /// A reader that always returns a fixed number of bytes counting up how many times `read` was
```

Comment:
This makes it look like the reader as whole is limited by the `usize` but it's not. (There's no `self.1 -= len` on purpose.)

The comment should say "returns at most `self.1` number of bytes from each `read` but is otherwise unlimited"